### ISSUE TITLE
setColumns loading a table with a new dataset

### ISSUE THREAD

**Issue by bobberry** [olifolkerd/tabulator#14]
Hello,
I am trying to reload a table with a new dataset. Meaning new column names (and a different number of them) along with new data. 
 I do tabulator("setColumns", newhdrs, false)  and then tabulator("setData", newdata)  table blanks out. Though I can tell by scroll bars that it has the new data. (by "blanks out" the header is empty, and that data area is empty except for the alternate row back ground color)

I tried tabulator("clear")  before changing - same thing.  I tried tabulator("setData", []) - same thing.
any hints?
Cheers
Bob

**Comment by bobberry** [comment]
Hello,
I assumed if I posted an Issue, that I would find my error within minutes - and it took 7 minutes.
The first time I init the tabulator the object contains {height:"320px", fitColumns:true, columns:{ col config} } so I just recreated that and passed it to "setColumns" instead of just the {col config} array of col objects.

Works Great - Thanks!
Cheers,
Bob.

**Comment by bobberry** [comment]
Hello,
But since I have already bothered you. How would you delete the tabulator and start fresh on the same page?  tabulator("destroy") does not seem to do anything.
cheers,
Bob.

**Comment by olifolkerd** [comment]
Hi Bobberry,

Good spot on the destroy function, i have just uploaded a fix, and that should work for you now.

let me know if there are any other features you would like to see in tabulator.

Cheers

Oli

**Comment by bobberry** [comment]
Hello Oli,Is there anyway to add a stand check box icon to a column?  I currently have the column set to formatter:"tickCross" - very colorful, but does not look like a standard checkbox.
awesome work btw - great data grid element.
cheers,Bob  Berry

 

    On Monday, February 8, 2016 2:45 PM, Oli Folkerd <notifications@github.com> wrote:
 

 Closed #14.—

**Comment by olifolkerd** [comment]
Hi Bob,

If you want to display a custom image in the column, I suggest using a custom formatter.

```'js
var customFormatter = function(value, data, cell, row, options, formatterParams){
        //value - the value of the cell
        //data - the data for the row the cell is in
        //cell - the DOM element of the cell
        //row - the DOM element of the row
        //options - the options set for this tabulator
        //formatterParams - parameters set for the column
        
        if(value){
                return "<img src='/path/to.img'/>"; // must return the html or jquery element of the html for the contents of the cell;
        }else{
                return "<img src='/path/to.img'/>"; // must return the html or jquery element of the html for the contents of the cell;
        }
        
    },
```
You can then assign the formatter to the column, in the column definition array
```js
{title:"Likes Cheese", field:"cheese", formatter:customFormatter},
```
If you wanted to display an actual tick box then you could return a tickbox element instead of an image (though i would make it read-only for consistency):
```js
return "<input type='checkbox' checked readonly />"
```

If you wanted the tickbox to to be editable, there is an editor already available for this called ***tick***, you would just need to assign it in the column definition array:
```js
{title:"Likes Cheese", field:"cheese", formatter:customFormatter, editor:"tick"},
```

Full info on Formatters can be found [here](https://github.com/olifolkerd/tabulator/wiki/Formatting-Data)

Full Info on Editors can be found [here](https://github.com/olifolkerd/tabulator/wiki/Manipulating-Data)

Cheers

Oli :)
